### ISSUE TITLE
Ensure that  vpub_in and vpub_out are not both nonzero

### ISSUE THREAD

**Issue by ebfull** [zcash/zcash#466]
Pour A values:

```
vpub_in = 5
oldcoin_0 = 0
oldcoin_1 = 0
newcoin_0 = 1
newcoin_1 = 0
vpub_out = 4
```

Pour B values:

```
vpub_in = 1
oldcoin_0 = 0
oldcoin_1 = 0
newcoin_0 = 1
newcoin_1 = 0
vpub_out = 0
```

Do these pours have the same semantics? They both validate and both affect the "value pool" in the exact same way. One of them just requires the value pool has more value in it than is really necessary.

Perhaps this should be a well-formedness check on pours if it has anonymity implications? If not, we can add this constraint later in a softfork.

**Comment by daira** [comment]
It's a potential distinguisher between clients, so a client should only generate Pours with either v<sup>old</sup><sub>pub</sub> = 0 or v<sup>old</sup><sub>pub</sub> = 0 or both, to maximise indistinguishability. If there is such an implicit rule, then Pours that don't follow it will stick out, so I think we should reconsider enforcing it as a consensus rule for 1.0. I can't see any harm in doing so, and it is very simple.

**Comment by daira** [comment]
Note that the Mint/Pour unification could alternatively have been expressed in terms of allowing negative v<sup>new</sup><sub>pub</sub>. I think the only reasons we didn't do that were avoiding signed authentic in the circuit (this was when we weren't as confident about circuit changes; it would actually have been simple), and some aesthetic symmetry between Pour inputs and outputs.

**Comment by ebfull** [comment]
You're right @daira. I wish we could go back and fix that. I do think analysis of the code surrounding pours was simpler when all the values were positive, though.

**Comment by daira** [comment]
This is in the current spec.

**Comment by daira** [comment]
Oh wait, is this ticket supposed to be for implementing the restriction, not just specifying it?

Note that it doesn't need to be enforced in the circuit, because v<sup>old</sub><sub>pub</sub> and v<sup>new</sub><sub>pub</sub> are public.

**Comment by nathan-at-least** [comment]
Closed by #897
